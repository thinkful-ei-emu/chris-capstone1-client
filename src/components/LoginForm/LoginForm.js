import React from 'react';
import TokenService from '../../services/token-service';
import { Button, Input } from '../Utils/Utils'

export default class LoginForm extends React.Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }

    state = {
        error: null
    }

    handleSubmitBasicAuth = ev => {
        ev.preventdefault()
        const {user_name, password } = ev.target

        TokenService.saveAuthToken(
            TokenService.makeBasicAuthToken(user_name.value, password.value)
        )

        user_name.value = '';
        password.value = '';
        this.props.onLoginSuccess()
    }

    render() {
        const { error } = this.state;
        return (
            <form
                className='LoginForm'
                onSubmit={this.handleSubmitBasicAuth}
            >
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className='user_name'>
                    <label htmlFor='LoginForm_user_name'>
                        Username
                    </label>
                    <Input required
                        name='user_name'
                        id='LoginForm_user_name'/>
                </div>
                <div className='password'>
                    <label htmlFor='LoginForm_password'>
                        Username
                    </label>
                    <Input required
                        name='password'
                        id='LoginForm_password'/>
                </div>
                <Button type='submit'>
                    Login
                </Button>
            </form>
        )
    }
}