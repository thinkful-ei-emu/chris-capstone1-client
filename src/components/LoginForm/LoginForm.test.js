import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';
import ReactDom from 'react-dom';

describe('LoginForm component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <LoginForm />,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes LoginForm by default', () => {
        expect(toJson(shallow(<LoginForm />))).toMatchSnapshot() 
    })
})