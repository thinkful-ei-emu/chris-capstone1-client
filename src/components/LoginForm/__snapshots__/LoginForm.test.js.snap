// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renderes LoginForm by default 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <Input
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <Input
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <Button
    type="submit"
  >
    Login
  </Button>
</form>
`;
